sandy/decay.py

```diff
@@ -148,6 +156,32 @@ def get_decayconstant(self, with_uncertainty=True):
         else:
             return df.LAMBDA
 
+    def custom_perturbation_decayconstant(self, pert):
+        """
+        Apply a custom perturbation to the decay constant.
+
+        Parameters
+        ----------
+        pert : `float`
+            Perturbation coefficient as ratio value.
+
+        Returns
+        -------
+        `sandy.decay.DecayData`
+            DacayData instance with given value ZAM and decay constant perturbed.
+
+        Examples
+        --------
+        >>> endf6 = sandy.get_endf6_file("jeff_33", 'decay', [922350, 922330, 922350])
+        >>> rdd = sandy.DecayData.from_endf6(endf6)
+        >>> dc_pert = sandy.DecayData.custom_perturbation_decayconstant(rdd, 0.05)
+        >>> print(dc_pert)
+        {922330: 1.4482087325525418e-13, 922350: 3.2768932273817556e-17}
+        """
+        dc = self.get_decayconstant().LAMBDA
+        dc = dc * (1 + pert)
+        return self.__class__(dc.to_dict())

```

Comment:
this cannot work!!!...you see it in your example.
When you `print(dc_pert)` - which is supposed to be a `DecayData` instance - you show the perturbed decay constants. Now, where are the decay energies? the spin? the half-life?....if you don't have them then your `DecayData` is not correct.

in theory we should add some `if` clauses in the `data.setter` of `DecayData` to make sure you are initializing correctly the instance.
@AitorBengoechea , what do we do for FYs?